### PR TITLE
Feature 911 tc trackint

### LINKED ISSUE #911: New Use Case: TC Verification Compare ADECK vs BDECK
## Describe the New Use Case ##
The National Hurricane Center organizes their track data into separate files for each storm. A historical archive of this data is available via the NHC ftp site:
https://ftp.nhc.noaa.gov/atcf/archive/

For past hurricane evaluations, the DTC ran the MET tc-pairs tool once for each input storm. METplus team members met with Yan Jin from the NOAA/HAFs group on 5/4/21. She is also processing data from NHC organized in this way. Recommend that we add a new use case to demonstrate the this type of processing logic.

The user defines a 4 digit year (e.g. 2019) and the basin(s) they'd like to process (AL or EP)? We could also have them explicitly specify a list of storm numbers to process. For example, there were 20 storms in the AL basin in 2019, numbered 01 to 20. This use case should loop through the storm numbers listed. For each, identify the adeck track (e.g. aal072019.dat for storm number 07), the bdeck track (e.g. bal072019.dat), and the edeck track (e.g. eal072019.dat). Then call tc_pairs passing these single files as input.

We probably do not need to process all 20 AL storms for this use case. Instead, consider processing 3 AL storms and 3 EP storms. And we probably don't need to process all available forecast tracks. We could limit it to something like: OFCL, HWRF, HMON, and GFSO.

Ideally, after processing all the storms in the list, we would then call METplotpy to plot the track and intensity errors for those storms. However, that functionality may not yet exist. We could also consider calling METdbload to load the .tcst output files into a MET database.

### Acceptance Testing ###
Recommend having DTC hurricane scientists evaluate this use case (@KathrynNewman, @mrinalbiswas, or @evankalina) to confirm that it's working as expected.

Recommend asking Yan Jin to also test this out.
*List input data types and sources.*
*Describe tests required for new functionality.*

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the new feature down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**
- [ ] Select **privacy**

### Projects and Milestone ###
- [ ] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [ ] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## New Use Case Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by mrinalbiswas**
## Pull Request Testing ##

- [ ] Describe testing already performed for these changes:</br>
Ran the use case on seneca. Got reasonable outputs. 

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [ ] Do these changes include sufficient testing updates? **[Yes]**

- [ ] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by georgemccabe** [comment]
A few notes about test differences that I mentioned in an email but should put here for traceability.

* The pull request run for your new use case should "fail" with a red X because the difference tests failed due to the new data from your use case being added. This is expected.
* The s2s:5 use case group will also fail. This is due to something that broke in the METplotpy repository that is being fixed. You can ignore this one.

The addition of new output will require an update to the develop-ref branch, however we need to wait until the changes to METplotpy are added to fix the broken s2s use cases. This PR can be merged once approved and I will handle the update of the truth data once the other use cases are fixed.

**Review by j-opatz — APPROVED** [review]
The use case runs successfully, with the only failing automated testing being the ones that were discussed (unrelated S2S, and new output data for this use case). Documentation is fine and develop data was updated by George. This PR is approved.

**Pull request merged**
